x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteForecastAction.java

```diff
@@ -249,7 +249,17 @@ private static void handleFailure(Exception e, DeleteForecastAction.Request requ
                 );
             }
         } else {
-            listener.onFailure(new ElasticsearchException("An error occurred while searching forecasts to delete", e));
+            if (e instanceof ElasticsearchException) {
+                listener.onFailure(
+                    new ElasticsearchStatusException(
+                        "An error occurred while searching forecasts to delete",
+                        ((ElasticsearchException) e).status(),
```

Comment:
You can use `instanceof` pattern matching to avoid the later cast

```suggestion
            if (e instanceof ElasticsearchException elasticsearchException) {
                listener.onFailure(
                    new ElasticsearchStatusException(
                        "An error occurred while searching forecasts to delete",
                        elasticsearchException.status(),
```

Reply:
awesome!